### ISSUE TITLE
POWER9 optimization?

### ISSUE THREAD

**Issue by madscientist159** [nioroso-x3/xmrig#1]
Are there any plans to optimize for POWER9?  On a single 3.8GHz 18 core device I can get a peak hash/watt of around 2.6 (corresponding hash/core 23.5), but this is still not profitable even with relatively low electric power rates.  Peak hash/core is much higher, something like 70H/core, but the H/w for that setting is laughably awful.

For comparison it seems the highly tuned algorithms for AMD (arguably backdoored) CPUs yield more in the range of 5H/w, which could still be (barely) profitable under some circumstances.

Would remote access to a dedicated POWER9 box help you at all?

**Comment by nioroso-x3** [comment]
There's not much else to optimize, now the JIT is as fast as compiled code, you can test it with the program in this pastebin: https://pastebin.com/dpJ7hpnL
CN/R did a good job on removing all advantages of a big L3.

**Comment by madscientist159** [comment]
Agreed, it does seem that CN/R is far more focused on GPU mining than anything else.  Still, POWER might make a good secure host for e.g. a ROCm-based miner -- any interest in working that angle if we provide access to sample hardware?

**Comment by nioroso-x3** [comment]
You want help to port xmrig-amd? Sure, if you have a machine with a power9 and some vegas available I could try, as long as I can use it to mine. Reminder though I like to do this in my free time, it might take a month or two.

**Comment by madscientist159** [comment]
> You want help to port xmrig-amd? Sure, if you have a machine with a power9 and some vegas available I could try, as long as I can use it to mine. Reminder though I like to do this in my free time, it might take a month or two.

Sure, no problem....I think I've got a dev box available with P9 and a Vega, main question is whether ROCm is actually working yet or not.

**Comment by nioroso-x3** [comment]
Well, you can test it using these scripts, looks like it pulls rocm directly from github and compiles it.
https://github.com/RadeonOpenCompute/Experimental_ROC

**Comment by nioroso-x3** [comment]
So did you get it working? I doubt AMD will want to help, since IBM partered with nvidia with their nvlink stuff.
